book/security.md

```diff
@@ -38,732 +43,781 @@ personalization or show you your private data.
-our hard-coded guest-book entry to the guest-book server. And
-remember: the form is submitted by the user's browser, with the user's
-cookies, so the post will succeed (if the user is logged in to the
-guest book). Posting a comment this way is not too scary (though shady
-advertisers will pay for it!) but posting a bank transaction is.
+Now this form can't be submitted except from our website---and if you
+repeat this nonce fix for each form in the web application, it'll be
+secure from CSRF attacks. But as with cross-site scripting attacks,
+it'd be better for the browser to provide a fail-safe solution that
+servers could leverage.
 
-Try it! This should work in both ours and in a real browser. Plus, in a
-real browser you could also have a "hidden" input element, which would
-mean that you don't need to write any tricky JavaScript at all.
+SameSite Cookies
+================
 
-How do we defend against this attack?
+For form submissions, that fail-safe solution is `SameSite` cookies.
+The idea is that a server can opt into its cookies being `SameSite`,
+which means they're not sent on cross-site form
+submissions.[^in-progress]
 
-We want to make sure that every POST request to `/add` comes from a form
-on our website. The normal way to ensure that is to embed a secret
-value, called a *nonce*, into the form, and to reject form submissions
-that don't come with this secret value.
+[^in-progress]: At the time of this writing, `SameSite` cookies are
+    not standardized, and different browsers handle them differently.
+    The [MDN page][mdn-samesite] is helpful for checking the current
+    status. Over time, this section may become out of date, though we
+    expect the general mechanism of `SameSite` cookies to survive.
+    
+[mdn-samesite]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#secure
 
-A nonce is like a login token, but instead of getting it on login, you
-get a new one every time you're presented with a form to fill out. In
-some sense, it's like a cookie, except that it's stored inside the
-HTML and thus lasts for just one request (instead of across many
-requests to the same server), so it can't be misused in the same way.
+A cookie is marked `SameSite` in the `Set-Cookie` header like this:
 
-Let's implement nonces in our guest book. Generate a secret nonce and
-add it to the form:
+    Set-Cookie: foo=bar; SameSite=Lax
 
-``` {.python expected=False}
-def show_comments(username):
-    # ...
-    if username:
-        nonce = str(random.random())[2:]
-        out += "<form action=add method=post>"
-        out +=   "<p><input name=nonce type=hidden value=" + nonce + "></p>"
-        # ...
+The `SameSite` attribute can take the value `Lax`, `Strict`, or
+`None`, and as I write this browsers have and plan different defaults.
+Our browser will default to `None` and implement `Lax` as an option.
+When `SameSite` is set to `Lax`, the cookie is not sent on cross-site
```

Comment:
Did you define same site?
https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-05#section-5.2

says:

A request is "same-site" if its target's URI's origin's registrable
   domain is an exact match for the request's client's "site for
   cookies", or if the request has no client.  The request is otherwise
   "cross-site".

We should also note that this behavior corresponds to how cookies have a different definition than the same-origin policy.

Reply:
Ok it seems you're right.